examples/tutorials/logging.ipynb

```diff
@@ -0,0 +1,351 @@
+   "execution_count": null,
+   "id": "cf576cf0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!cat hello-numpy-sag-workspace/server/log.json"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bb3436da",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!cat hello-numpy-sag-workspace/server/log_error.txt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "526434bf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!cat hello-numpy-sag-workspace/server/log_fl.txt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d30ee805",
+   "metadata": {},
+   "source": [
+    "### Customization\n",
+    "\n",
+    "The log config file can easily be customized for different use cases.\n",
+    "\n",
+    "For this example, let's assume we are a federated learning researcher mainly interested in the algorithm parts of the log.\n",
+    "By default, we provide the FLFileHandler which uses the FLFilter to generate the log_fl.txt However we would also like to have a cleaner output in the console, as well as remove/shorten some unnecessary log fields.\n",
+    "\n",
+    "### custom_log_config.json\n",
+    "\n",
+    "Let's change the consoleHandler accordingly to achieve our goal of a cleaner console output.\n",
+    "\n",
+    "Remove the fl_ctx field, and add the identity field for a cleaner log structure:\n",
+    "\n",
+    "`\"fmt\": \"%(asctime)s - %(identity)s - %(name)s - %(levelname)s - %(message)s\",`\n",
+    "\n",
+    "Configure the datefmt to only use seconds rather than milliseconds:\n",
+    "\n",
+    "``\"datefmt\": \"%Y-%m-%d %H:%M:%S\"``\n",
+    "\n",
+    "Add the FLFilter, which only allows logs related to FL training to pass through:\n",
+    "\n",
+    "``\"filters\": [\"FLFilter\"],``\n",
+    "\n",
+    "Say we want to highlight the NPModelPersistor, we can color it blue to make it stand out in the console:\n",
+    "```\n",
+    "\"logger_colors\": {\n",
+    "    \"NPModelPersistor\": \"blue\"\n",
+    "}\n",
+    "```\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cea18ebc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!cat custom_log_config.json"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b6caecd4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!diff custom_log_config.json hello-numpy-sag-workspace/server/local/log_config.json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2d563228",
+   "metadata": {},
+   "source": [
+    "Now let's run the simulator with the custom_log_config.json using the ``-l`` option:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4b6e2cd8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!nvflare simulator -w hello-numpy-sag-workspace -n 2 -t 2 -l custom_log_config.json ../hello-world/hello-numpy-sag/jobs/hello-numpy-sag"
```

Comment:
For simulator, do all the servers and clients share the same log_config?

Reply:
For the `-l` argument, yes for simplicity. If they really wanted to change it for each site they could do so in the workspace